src/Serilog.Sinks.Datadog.Logs/Sinks/Datadog/DatadogSink.cs

```diff
@@ -53,6 +65,25 @@ public DatadogSink(string apiKey, string source, string service, string host, st
             _client = new DatadogClient(config);
         }
 
+        private string formatMessage(LogEvent logEvent, string source, string service, string host, string tags) {
+            var payload = new StringBuilder();
+            var writer = new StringWriter(payload);
+
+            // Serialize the event as JSON. The Serilog formatter handles the
+            // internal structure of the logEvent to give a nicely formatted JSON
+            formatter.Format(logEvent, writer);
+
+            // Convert the JSON to a dictionnary and add the DataDog properties
+            var logEventAsDict = JsonConvert.DeserializeObject<Dictionary<string, dynamic>>(payload.ToString());
+            logEventAsDict.Add("ddsource", source);
```

Comment:
What happens if the entry already exists ? What should we do ?

Reply:
The object we serialize and deserialize is a LogEvent. The keys that this object have are controlled: https://github.com/serilog/serilog/blob/9a46e68d3e224b965f4ac2665fd379ae9568d6b4/src/Serilog/Formatting/Json/JsonFormatter.cs#L117-L157
They also are capitalized, so we don't risk any collision

Reply:
👍